src/connector.ts

```diff
@@ -53,30 +55,41 @@ export declare interface DriverOptions {
 class CloudSQLInstanceMap extends Map {
   async loadInstance({
     ipType,
+    authType,
     instanceConnectionName,
     sqlAdminFetcher,
   }: {
     ipType: IpAddressTypes;
+    authType: AuthTypes;
     instanceConnectionName: string;
     sqlAdminFetcher: SQLAdminFetcher;
   }): Promise<void> {
     // in case an instance to that connection name has already
     // been setup there's no need to set it up again
     if (this.has(instanceConnectionName)) {
       const instance = this.get(instanceConnectionName);
-      if (instance.ipType !== ipType) {
+      if (instance.ipType && instance.ipType !== ipType) {
         throw new CloudSQLConnectorError({
           message:
             `getOptions called for instance ${instanceConnectionName} with ipType ${ipType}, ` +
             `but was previously called with ipType ${instance.ipType}. ` +
             'If you require both for your use case, please use a new connector object.',
           code: 'EMISMATCHIPTYPE',
         });
+      } else if (instance.authType && instance.authType !== authType) {
```

Comment:
```suggestion
      } else if (instance.authType !== authType) {
```

similar to the previous comment